src/main/java/appeng/thirdparty/codechicken/lib/model/pipeline/transformers/QuadCornerKicker.java

```diff
@@ -0,0 +1,224 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+package appeng.thirdparty.codechicken.lib.model.pipeline.transformers;
+
+
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.EnumFacing.AxisDirection;
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraft.util.math.Vec3i;
+
+import appeng.thirdparty.codechicken.lib.model.Quad.Vertex;
+import appeng.thirdparty.codechicken.lib.model.pipeline.IPipelineElementFactory;
+import appeng.thirdparty.codechicken.lib.model.pipeline.QuadTransformer;
+
+import static net.minecraft.util.EnumFacing.AxisDirection.NEGATIVE;
+import static net.minecraft.util.EnumFacing.AxisDirection.POSITIVE;
+
+
+/**
+ * This transformer is a little complicated.
+ * Basically a Facade / Cover can use this to 'kick' the edges
+ * in of quads to fix z-Fighting in the corners.
+ * Use it by specifying the side of the block you are on,
+ * the bitmask for where the other Facades / Cover's are,
+ * the bounding box of the facade, NOT the hole piece,
+ * and the thickness of your Facade / Cover, this is used
+ * as the kick amount.
+ *
+ * @author covers1624
+ */
+public class QuadCornerKicker extends QuadTransformer
+{
+
+	//The factory for pipeline creation.
+	public static final IPipelineElementFactory<QuadCornerKicker> FACTORY = QuadCornerKicker::new;
+
+	//Simple horizonal lookups.
+	public static int[][] horizonals = new int[][] {
+			//Around Y axis, NSWE.
+			{ 2, 3, 4, 5 },//
+			{ 2, 3, 4, 5 },//
+
+			//Around Z axis, DUWE.
+			{ 0, 1, 4, 5 },//
+			{ 0, 1, 4, 5 },//
+
+			//Around X axis, DUNS.
+			{ 0, 1, 2, 3 },//
+			{ 0, 1, 2, 3 }
+	};
+
+	private int mySide;
+	private int facadeMask;
+	private AxisAlignedBB box;
+	private double thickness;
+
+	QuadCornerKicker()
+	{
+		super();
+	}
+
+	/**
+	 * Set's the side this Facade / Cover is attached to.
+	 *
+	 * @param side The side.
+	 */
+	public void setSide( int side )
+	{
+		this.mySide = side;
+	}
+
+	/**
+	 * Sets the bitmask of Facades / Covers in the blockspace.
+	 * This is as simple as, mask = (1 << side)
+	 *
+	 * @param mask The mask.
+	 */
+	public void setFacadeMask( int mask )
+	{
+		this.facadeMask = mask;
+	}
+
+	/**
+	 * Sets the bounding box of the Facade / Cover,
+	 * this should be the full box, not just a piece
+	 * of the hole's 'ring'.
+	 *
+	 * @param box The BoundingBox.
+	 */
+	public void setBox( AxisAlignedBB box )
+	{
+		this.box = box;
+	}
+
+	/**
+	 * Sets the amount to kick the vertex in by,
+	 * this is your facades thickness.
+	 *
+	 * @param thickness The thickness.
+	 */
+	public void setThickness( double thickness )
+	{
+		this.thickness = thickness;
+	}
+
+	@Override
+	public boolean transform()
+	{
+
+		int side = quad.orientation.ordinal();
+		if( side != mySide && side != ( mySide ^ 1 ) )
+		{
+			for( int hoz : horizonals[mySide] )
+			{
+				if( side != hoz && side != ( hoz ^ 1 ) )
+				{
+					if( ( facadeMask & ( 1 << hoz ) ) != 0 )
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS134)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1066)